lib/chef/resource.rb

```diff
@@ -416,6 +420,10 @@ def only_if(command = nil, opts = {}, &block)
     # @param block [Proc] A ruby block to run. Ignored if a command is given.
     #
     def not_if(command = nil, opts = {}, &block)
+      if command.is_a?(Hash) && opts.empty?
+        opts = command
+        command = nil
+      end
```

Comment:
does `def not_if(command = nil, **opts, &block)` work here without this mangling, because this code seems brittle to me...